actionlib_msgs/package.xml

```diff
@@ -20,6 +20,8 @@
 
   <test_depend>ament_lint_common</test_depend>
 
+  <member_of_group>interface_packages</member_of_group>
```

Comment:
Is this name set in a document somewhere? Or are we coming up with it for the first time here? If it's not too hard to change yet, I'd slightly prefer `rosidl_interface_packages`, since there might be other kinds of interfaces in the future that might want to make use of the group mechanism.

Reply:
I'm fine with either `interface_packages` or `rosidl_interface_packages` as William suggests.

Reply:
this name is not used or documented anywhere yet, the goal of this PR was to get feedback on what name we think makes most sense before starting using it.
In ROS 1 we abuse the term "message package" as sometimes we refer to packages containing services or actions definitions, that's why I used `interfaces` instead.
One thing to keep in mind is that ROS 1 packages migrating to format 3 may want to use something similar. The resulting question is "should the group name be the same in ROS 1 and ROS 2?" If no I agree that `rosidl_interface_packages` would be better than `interface_packages` , if yes we need a name that is generic enough to make sense for both ROS 1 and ROS 2 (that could be `rosidl_interface_packages` or something else, not sure yet)

Reply:
Another question is, do we want/need different groups for packages which have messages, services, and actions?

Reply:
Indeed that's another good question.
If the answer is yes, this would also raise a few more points:
- our tools should be able to support overlapping groups / duplicated dependencies (already the case I think)
- should we support metagroups? e.g. interface_packages group depending on groups message_packages + services_packages + action_packages

Reply:
As for a name that's appropriate for both ROS 1 and ROS 2, first I think `rosidl` and `interface` are generic enough for both, though you're right that in ROS 1 it was just as common to call them just "messages" or associate them with the "genmsg". Second, I believe the groups can use conditionals as well, so even if they needed to be different in ROS 1 and ROS 2 with the same `package.xml` then you could choose which group to use in each case.

Reply:
> Second, I believe the groups can use conditionals as well, so even if they needed to be different in ROS 1 and ROS 2 with the same package.xml then you could choose which group to use in each case.

I was mostly wondering if we think they _should_ be different or not. If they end up being different we can use conditions :+1:

Reply:
What's the benefit of having different groups for messages, services, and actions? Is it planned to do code generation for them in separate packages?

The `interface_packages` groups seems special since it will have so many members in core packages. I think it would be a good idea to adopt a prefix to names like this. It avoids name collisions with third-party groups. Also, googling `ros_interface_packages` versus `interface_packages` might be more likely to return a search result to a page describing what the group is.